src/libertem/api.py

```diff
@@ -223,12 +231,14 @@ def __init__(
         self,
         executor: Optional[JobExecutor] = None,
         plot_class: Optional['Live2DPlot'] = None,
+        main_process_gpu: Optional[int] = None,
     ):
         import traceback
         if executor is None:
             executor = self._create_local_executor()
         self.executor = executor
         self._plot_class = plot_class
+        self._main_process_gpu = main_process_gpu
```

Comment:
A quick question, I might be missing something technical, but does this need to live on the `Context` itself? It seems that so far the details of *how* the computation happens are confined to the various Executors. So `main_process_gpu` could be passed into the executor creation by `make_with`, then the executor itself handles supplying the GPU `xp` during any main process code? (Or the executor directly created with main node GPU enabled, of course, and the Context would not need to know).

Reply:
That's a good point! The executor is currently not involved in running the main process part, only the Tasks on the workers. The UDFRunner is ephemeral. Going up the call stack I ended up at the Context to store the state and feed the parameter into the UDFRunner.

If one wanted to let an Executor handle this, one could perhaps use a second one, for example a Threaded one, for the main process? That would also allow running the merge and results functions of several UDFs in parallel in the future.

I'll see how much work it would be to implement a simple version of this, just to get the API right -- `__init__()` should accept a second executor in that case.

I think it would mean adding GPU support to the executor, I'll see what would be the best way for this.

Reply:
Looking at it a little, could we do something like using the executor itself to call `runner.run_for_dataset_sync`, with something like an `executor.run_main_process` and so it could apply the `use_gpu` context manager as appropriate to its setting?